### PR TITLE
Ria 3600 request review notification

### PR DESCRIPTION
**Pull request by jkumar-moj**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3600

### Change description ###
HomeOffice notification for request review event

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestEvidenceBundleNotificationHandler.java [hunk 1]
```diff
@@ -52,58 +53,36 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCa
...
+        log.info("Finished constructing {} notification request for caseId: {}, HomeOffice reference: {}",
+            REQUEST_EVIDENCE_BUNDLE.toString(), caseId, homeOfficeReferenceNumber);
 
-        homeOfficeInstructService.extractDirectionAttributes(
-            messageBuilder, asylumCase, DirectionTag.RESPONDENT_EVIDENCE);
+        final String notificationStatus = homeOfficeInstructService.sendNotification(bundleInstructMessage);
 
-        HomeOfficeChallenge challenge = homeOfficeInstructService.buildHomeOfficeChallenge(asylumCase);
-        messageBuilder.withChallenge(challenge);
+        asylumCase.write(AsylumCaseDefinition.HOME_OFFICE_INSTRUCT_STATUS, notificationStatus);
```
Comment (divisathyan) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestEvidenceBundleNotificationHandler.java:81]: status should be set for Request Evidence specifically

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/UuidIdentityProvider.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service;
+
+import java.util.UUID;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UuidIdentityProvider implements IdentityProvider {
+
+    public String identity() {
```
Comment (divisathyan) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/UuidIdentityProvider.java:9]: I think we are duplicating logic. We can reuse HomeOfficeRequestUuidGenerator.generateUuid()

**Review by divisathyan — COMMENTED** [review]
Please see comments

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/AsylumCaseDefinition.java [hunk 3]
```diff
@@ -44,7 +44,13 @@
         "directions", new TypeReference<List<IdValue<Direction>>>(){}),
 
     HOME_OFFICE_INSTRUCT_STATUS(
-        "homeOfficeInstructStatus", new TypeReference<String>() {});
+        "homeOfficeInstructStatus", new TypeReference<String>() {}),
```
[the remark this answers is not in the document]
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/AsylumCaseDefinition.java:47]: Eventually not. I haven't changed the validation yet. Just did not want to mix things. Will do it as a part of upcoming stories.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestEvidenceBundleNotificationHandler.java (hunk 1 above)
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestEvidenceBundleNotificationHandler.java:81]: See the above comment.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/UuidIdentityProvider.java (hunk 2 above)
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/UuidIdentityProvider.java:9]: I will get rid off the old implementation HomeOfficeRequestUuidGenerator in a separate cleanup task along with the INSTRUCT_STATUS flag. The above implementation (UuidIdentityProvider) was added as it gives us the ability to mock and verify for testing.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/AsylumCaseDefinition.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/AsylumCaseDefinition.java:47]: ok

**Review by divisathyan — APPROVED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Comment by jkumar-moj** [comment]
Reopening as the PR was closed by stale bot

**Pull request merged**